Documentation: nRF Desktop multi DFU (#11173)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/dfu.rst

   By default, the reported Vendor ID, Product ID, and generation are aligned with the values defined globally for the nRF Desktop application.
   The default values of Kconfig options used by the DFU module are based on respectively :ref:`CONFIG_DESKTOP_DEVICE_VID <config_desktop_app_options>`, :ref:`CONFIG_DESKTOP_DEVICE_PID <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_DEVICE_GENERATION <config_desktop_app_options>`.

Flash access synchronization with other DFU methods
===================================================

The DFU module leverages the :ref:`nrf_desktop_dfu_lock` to synchronize flash access with other DFU methods (e.g. SMP DFU).
Please use the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option to activate this behaviour.

On each DFU attempt, the module attempts to claim ownership over the DFU flash using the DFU Lock API.

> It holds the DFU owner status until the DFU process is complete or timed out.

Reviewer: ```suggestion
It holds the DFU owner status until the DFU process is completed or timed out.
```
since we're talking about process (unified in the other place as well)

   By default, the reported Vendor ID, Product ID, and generation are aligned with the values defined globally for the nRF Desktop application.
   The default values of Kconfig options used by the DFU module are based on respectively :ref:`CONFIG_DESKTOP_DEVICE_VID <config_desktop_app_options>`, :ref:`CONFIG_DESKTOP_DEVICE_PID <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_DEVICE_GENERATION <config_desktop_app_options>`.

Flash access synchronization with other DFU methods
===================================================

The DFU module leverages the :ref:`nrf_desktop_dfu_lock` to synchronize flash access with other DFU methods (for example, SMP DFU).

> Other DFU methods should also use the :ref:`nrf_desktop_dfu_lock` for the synchronization mechanism to be effective.

Reviewer 2: ```suggestion
Other DFU methods should also use the :ref:`nrf_desktop_dfu_lock` for an effective synchronization mechanism.
```

Author: fixed

> Make sure that the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option is enabled if multiple DFU transports are enabled in the application configuration.

Author: rephrased it a bit

> If multiple DFU transports are enabled in your application configuration, make sure that the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option is enabled and all of the used DFU transports use the the :ref:`nrf_desktop_dfu_lock`.

Author: fixed

On each DFU attempt, the module attempts to claim ownership over the DFU flash using the DFU Lock API.
It holds the DFU owner status until the DFU process is completed or timed out.
The lock is also kept during the background erase operation as it changes the DFU flash memory.

> If the module is not the current DFU owner, it desists from performing requested actions that either write to the DFU flash or reboot the device.

Reviewer 3: `writes ... or reboots` ?

Author: actions = plural

> The module also performs the background erase operation of the DFU flash once it becomes the lock owner after another DFU method.

Reviewer 3: `performs` ->`automatically starts`

   The default values of Kconfig options used by the DFU module are based on respectively :ref:`CONFIG_DESKTOP_DEVICE_VID <config_desktop_app_options>`, :ref:`CONFIG_DESKTOP_DEVICE_PID <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_DEVICE_GENERATION <config_desktop_app_options>`.

Flash access synchronization with other DFU methods
===================================================

The DFU module leverages the :ref:`nrf_desktop_dfu_lock` to synchronize flash access with other DFU methods (for example, SMP DFU).
If multiple DFU transports are enabled in your application configuration, make sure that the following conditions are met:

* The :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option is enabled
* All of the used DFU transports use the :ref:`nrf_desktop_dfu_lock` module.

On each DFU attempt, the module attempts to claim ownership over the DFU flash using the DFU Lock API.
It holds the DFU owner status until the DFU process is completed or timed out.
The lock is also kept during the background erase operation as it changes the DFU flash memory content.

> If the module is not the current DFU owner, it desists from performing requested actions that either modify the DFU flash content or reboot the device.

Reviewer 3: `... is not the current DFU owner and cannot claim the DFU lock ...`

===================================================

The DFU module leverages the :ref:`nrf_desktop_dfu_lock` to synchronize flash access with other DFU methods (for example, SMP DFU).
Other DFU methods must also use the :ref:`nrf_desktop_dfu_lock` for an effective synchronization mechanism.
If multiple DFU transports are enabled in your application configuration, make sure that the following conditions are met:

* The :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option is enabled
* All of the used DFU transports use the :ref:`nrf_desktop_dfu_lock` module.

On each DFU attempt, the module attempts to claim ownership over the DFU flash using the DFU Lock API.
It holds the DFU owner status until the DFU process is completed or timed out.
The lock is also kept during the background erase operation as it changes the DFU flash memory content.

If the module is not the current DFU owner, it desists from performing requested actions that either write to the DFU flash or reboot the device.

The module also automatically starts the background erase operation of the DFU flash once it becomes the lock owner after another DFU method.

> 

Reviewer 3: We may also update the `nrf_desktop_ble_smp` references in documentation. Currently it would be better to use our application-specific module because it improves synchronization (may be done in a separate doc improvement PR).

Author: moved to follow-up

reboot
   Perform the fetch operation on this option to trigger an instant reboot of the device.

>    With the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option enabled, the device only executes this command when it holds the DFU owner status.

Reviewer 2: We could use a shortcut for the repeated text. 
For example: https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf_desktop/doc/battery_meas.rst#module-events

>    With the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option enabled, the device only executes this command when it holds the DFU owner status.

Reviewer 2: ```suggestion
   The device executes this command, only if the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option is enabled, and it holds the DFU owner status.
```
Hope the meaning has not changed.

Author: It has. With the DFU lock disabled, the command will execute no matter the DFU owner status.

```suggestion
   If the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option is enabled, the device executes this command, only if it holds the DFU owner status.
```
